File: translations/codenet/deepseek_commented/C++2Go/gpt-4o/Code/s992338735.go
package main

import (
	"fmt"
)

func main() {
	var N, S int // Declare variables

	for {
		_, err := fmt.Scan(&N) // Read the number of elements (N)
		if err != nil || N <= 0 {
			break // Exit the loop if input is invalid or N is non-positive
		}

		M := -1 // Initialize maximum
		m := 1001 // Initialize minimum
		addup := 0 // Initialize sum

		for i := 0; i < N; i++ { // Loop through each element
			fmt.Scan(&S) // Read the element
			addup += S // Update sum
			if S > M {
				M = S // Update maximum
			}
			if S < m {
				m = S // Update minimum
			}
		}

		// Calculate the average of the remaining numbers
		if N > 2 {
			fmt.Println((addup - M - m) / (N - 2))
		} else {
			fmt.Println(0) // If N <= 2, output 0 to avoid division by zero
		}
	}
} // End of the main function
// <END-OF-CODE>

File: translations/codenet/deepseek_commented/C++2P/granite/Code/s992338735.py

# Read the number of elements (N)
while True:
    N = int(input())
    if N <= 0:
        break
    # Initialize variables for maximum, minimum, sum, and index
    M = -1
    m = 1001
    addup = 0
    for i in range(N):
        # Read each element and update the maximum, minimum, and sum
        S = int(input())
        addup += S
        if S > M:
            M = S
        if S < m:
            m = S
    # Calculate the average of the remaining numbers
    print((addup - M - m) / (N - 2))

# End the loop and main function
print("